docs/docsite/rst/shared_snippets/variable_evaluation.rst

```diff
@@ -0,0 +1,30 @@
+.. _variable_evaluation:
+
+Variable Evaluation
+```````````````````
+
+In general, Ansible evaluates any variables in playbook content at the last
+possible second, which means that if you define a data structure that data
+structure itself can define variable values within it, and everything “just
+works” as you would expect. This also means variable strings can include other
+variables inside of those strings. This is known as Lazy Evaluation, which
+allows for variables to be defined out of order so long as they are defined
+before usage.
+
+::
+
+    vars:
+        a: "{{ b }}"
+        b: 3
+
+
+When using an Ansible :ref:`Lookup Plugin`, do note that the value will be
```

Comment:
This paragraph seems to conflate at least two important points which may merit separate paragraphs:
1) lazy template evaluation has the knock-on effect that lookup plugins may be executed more than once, take care if these are not idempotent or if they have side effects
2) set_fact is a mechanism by which you can "fix" template evaluation at a point in time